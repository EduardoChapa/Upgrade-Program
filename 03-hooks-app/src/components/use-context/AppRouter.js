import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route,
    // Link
} from "react-router-dom";
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NavBar from './NavBar';

function AppRouter() {
    return (
        <Router>
            <div>
                <NavBar />
                <Routes>
                    <Route path='/' element={ <HomeScreen /> } />
                    <Route path='*' element={ <HomeScreen /> } />
                    <Route path='/login' element={ <LoginScreen /> } />
                    <Route path='/about' element={ <AboutScreen /> } />
                </Routes>
            </div>
        </Router>
    );
}

export default AppRouter;
