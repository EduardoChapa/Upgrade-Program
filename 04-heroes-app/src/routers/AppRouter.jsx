import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import LoginScreen from '../components/login/LoginScreen';
import { NavBar } from '../components/ui/NavBar';
import DashboardRouter from './DashboardRouter';

function AppRouter() {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/login' element={ <LoginScreen /> }/>
                <Route path='/*' element={ <DashboardRouter /> } />
            </Routes>
        </BrowserRouter>
    )
}

export default AppRouter;
