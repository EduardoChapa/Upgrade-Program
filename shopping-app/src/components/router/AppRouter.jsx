import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import ShoppingApp from '../../ShoppingApp';
import ItemGrid from '../items/ItemGrid';
import HomeScreen from '../main/HomeScreen';
import ProfileScreen from '../profile/ProfileScreen';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path='/' element={ <ShoppingApp /> } >
          <Route path='/' element={ <HomeScreen /> } />
          <Route path='items' element={ <ItemGrid /> } />
          <Route path='profile' element={ <ProfileScreen /> } />
        </Route>
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;