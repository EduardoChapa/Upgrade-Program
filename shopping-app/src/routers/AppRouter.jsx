import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import ShoppingApp from '../ShoppingApp';
import Favorites from '../components/account/Favorites';
import OrderHistory from '../components/account/OrderHistory';
import Cart from '../components/cart/Cart';
import ItemGrid from '../components/items/ItemGrid';
import HomeScreen from '../components/home/HomeScreen';
import AccountScreen from '../components/account/AccountScreen';
import AccountDetails from '../components/account/AccountDetails';
import ItemScreen from '../components/items/ItemScreen';
import Items from '../components/items/Items';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path='/' element={ <ShoppingApp /> } >
          <Route path='*' element={ <HomeScreen /> } />
          <Route path='/' element={ <HomeScreen /> } />
          <Route path='items' element={ <Items /> } >
            <Route index element={ <ItemGrid /> } />
            <Route path=':productId' element={ <ItemScreen /> } />
          </Route>
          <Route path='account' element={ <AccountScreen /> } >
            <Route index element={ <AccountDetails /> } />
            <Route path='details' element={ <AccountDetails /> } />
            <Route path='orders' element={ <OrderHistory /> } />
            <Route path='favorites' element={ <Favorites /> } />
          </Route>
          <Route path='cart' element={ <Cart /> } />
        </Route>
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;