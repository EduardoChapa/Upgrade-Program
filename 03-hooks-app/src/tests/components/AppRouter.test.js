import React from 'react';
import { mount } from 'enzyme';
import AppRouter from '../../components/use-context/AppRouter';
import UserContext from '../../components/use-context/UserContext';

describe('Testing LoginScreen component', () => {

    const user = {
        id: 1,
        name: 'Ricardio'
    }

    const wrapper = mount(
        <UserContext.Provider value={{ user }} >
            <AppRouter />
        </UserContext.Provider>
    );

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});