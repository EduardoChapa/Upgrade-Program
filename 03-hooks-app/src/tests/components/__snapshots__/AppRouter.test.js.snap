// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginScreen component should display correctly 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
  <div>
    <h1>
      Home Screen
    </h1>
    <hr />
    <pre>
      {
   "id": 1,
   "name": "Ricardio"
}
    </pre>
  </div>
</div>
`;
